### ISSUE TITLE
Confusing error message when implementation lacks parameter default in overloaded function

### ISSUE THREAD

**Issue by shoffmeister** [python/mypy#11308]
**Bug Report**

mypy emits "Overloaded function implementation does not accept all possible arguments of signature 1" although "Any" should capture all possible signatures.

**To Reproduce**

```python
from typing import overload, AnyStr, Any

@overload
def thing(path: None = ...) -> str:
    ...
@overload
def thing(path: AnyStr) -> str:
    ...
def thing(path: Any) -> str:
    return type(path).__name__

print(thing(None))
print(thing("wombat"))
print(thing(b"wombat"))
```

**Expected Behavior**

No diagnostic.

**Actual Behavior**

"Overloaded function implementation does not accept all possible arguments of signature 1"

**Your Environment**

- Mypy version used: 0.9.10
- Python version used: 3.9.7

**Comment by erictraut** [comment]
Yes, I agree this is a false negative in pyright. It should emit an error in this case.

**Comment by shoffmeister** [comment]
> The reason mypy rejects this is not because of `Any`, but because the implementation doesn't specify that `path` is a parameter with a default. 

Understood, thank you :)

>The error message is quite misleading however.

The (changed) code below emits
-> Overloaded function implementation does not accept all possible arguments of signature 1
-> Overloaded function implementation does not accept all possible arguments of signature 2

Is that the same kind of misleading message?

As far as I understand it, the overloads only specify the contract which the implementation eventually will implement. But these two overloads are conflicting by themselves?

```
from typing import AnyStr, overload

@overload
def thing(path: bytes = b'a bytes') -> str: ...
@overload
def thing(path: str = 'a str') -> str: ...

def thing(path: AnyStr) -> str:
    return ""
